packages/@aws-cdk/core/lib/runtime.ts

```diff
@@ -218,8 +218,11 @@ export function validateString(x: any): ValidationResult {
 
 export function validateNumber(x: any): ValidationResult {
   if (canInspect(x) && typeof x !== 'number') {
-    return new ValidationResult(`${JSON.stringify(x)} should be a number`);
+    if (typeof x === 'string' && isNaN(parseInt(x))) {
```

Comment:
Why must it be an int? Is `"3.141592654"` not allowed? Why does `"10"` get preferential treatment?